src/include/duckdb/planner/operator/logical_positional_join.hpp

```diff
@@ -0,0 +1,33 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/planner/operator/logical_pos_join.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/planner/logical_operator.hpp"
+
+namespace duckdb {
+
+//! LogicalPositionalJoin represents a row-wise join between two relations
+class LogicalPositionalJoin : public LogicalOperator {
```

Comment:
I thought of that but it isn't really because it has no join type or conditions. My design was basically to create the unholy love child of cross join and table scan, so there are pieces from both but it really is its own beast.

Reply:
`LogicalCrossProduct` is not derived from `LogicalJoin` either.

Reply:
Perhaps it should inherit/share a base type with the `LogicalCrossProduct` then?